WooCommerce/src/androidTest/kotlin/com/woocommerce/android/ui/main/ProductsUITest.kt

```diff
@@ -0,0 +1,69 @@
+package com.woocommerce.android.ui.main
+
+import androidx.test.rule.ActivityTestRule
+import com.woocommerce.android.BuildConfig
+import com.woocommerce.android.helpers.InitializationRule
+import com.woocommerce.android.helpers.TestBase
+import com.woocommerce.android.screenshots.TabNavComponent
+import com.woocommerce.android.screenshots.login.WelcomeScreen
+import com.woocommerce.android.screenshots.products.ProductListScreen
+import com.woocommerce.android.screenshots.util.ProductData
+import com.woocommerce.android.screenshots.util.MocksReader
+import com.woocommerce.android.screenshots.util.iterator
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import org.json.JSONObject
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@HiltAndroidTest
+class ProductsUITest : TestBase() {
+    @get:Rule(order = 0)
+    val rule = HiltAndroidRule(this)
+
+    @get:Rule(order = 1)
+    val initRule = InitializationRule()
+
+    @get:Rule(order = 3)
+    var activityRule = ActivityTestRule(MainActivity::class.java)
+
+    @Before
+    fun setUp() {
+        WelcomeScreen
+            .logoutIfNeeded()
+            .selectLogin()
+            .proceedWith(BuildConfig.SCREENSHOTS_URL)
+            .proceedWith(BuildConfig.SCREENSHOTS_USERNAME)
+            .proceedWith(BuildConfig.SCREENSHOTS_PASSWORD)
+
+        TabNavComponent().gotoProductsScreen()
+    }
+
+    private fun mapJSONToProduct(productJSON: JSONObject): ProductData {
+        return ProductData(
+            id = productJSON.getInt("id"),
+            name = productJSON.getString("name"),
+        )
+    }
+
+    @Test
+    fun productListCheck() {
+        val productsJSONArray = MocksReader().readAllProductsToArray()
+
+        for (productJSON in productsJSONArray.iterator()) {
+            // parsing the json data
+            val productData = mapJSONToProduct(productJSON)
+
+            // create the screen ui object and select the product from the list by name
+            val productListScreen = ProductListScreen()
+                .scrollToProduct(productData.name)
+                .assertProductCardMatchesMockData(productData)
+
+            // for that detail view, check the data against the UI elements.
+            productListScreen.selectSingleProductByName(productData.name)
+                .assertProductDetails(productData)
+                .goBackToProductsScreen()
```

Comment:
One advantage the current screens structure provides us with, is the ability to use method chaining, and be free from the need to create constants:

```suggestion
            ProductListScreen()
                .scrollToProduct(productData.name)
                .assertProductCardMatchesMockData(productData)
                .selectSingleProductByName(productData.name)
                .assertProductDetails(productData)
                .goBackToProductsScreen()
```

Reply:
Done in 1f44767ec690b3689f718698a0a4b855e6fdbda7.